profile-crud/main.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/CollActionteam/collaction_backend/profileservice"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+func getProfile(req events.APIGatewayV2HTTPRequest) (events.APIGatewayProxyResponse, error) {
+	var msg profileservice.Response
+
+	profileData, err := profileservice.GetProfile(req)
+	if err != nil {
+		msg = profileservice.Response{
+			Message: "Error Retreiving Profile",
+			Data:    "",
+			Status:  http.StatusInternalServerError,
+		}
+		jsonData, _ := json.Marshal(msg)
+		return events.APIGatewayProxyResponse{
+			StatusCode: http.StatusInternalServerError,
+			Body:       string(jsonData),
+		}, err
+
+	}
+
+	if profileData == nil {
+		msg = profileservice.Response{
+			Message: "no user Profile found",
+			Data:    "",
+			Status:  http.StatusNotFound,
+		}
+		jsonData, _ := json.Marshal(msg)
+		return events.APIGatewayProxyResponse{
+			StatusCode: http.StatusNotFound,
+			Body:       string(jsonData),
+		}, nil
+	}
+
+	msg = profileservice.Response{
+		Message: "Successfully Retrieved Profile",
+		Data:    profileData,
+		Status:  http.StatusOK,
+	}
+	jsonData, _ := json.Marshal(msg)
+
+	return events.APIGatewayProxyResponse{
+		StatusCode: http.StatusOK,
+		Body:       string(jsonData),
+	}, err
+}
+
+func getProfileByID(req events.APIGatewayV2HTTPRequest) (events.APIGatewayProxyResponse, error) {
```

Comment:
This seems like a lot of duplicate code.
Can this be merged with the other get-function?